pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -403,8 +402,7 @@ func (p *ldapProvider) searchUser(name string, config *v3.LdapConfig, lConn *lda
 
 func (p *ldapProvider) searchGroup(name string, config *v3.LdapConfig, lConn *ldapv2.Conn) ([]v3.Principal, error) {
 	var groupScope string
-	query := "(&(" + config.GroupSearchAttribute + "=*" + name + "*)(" + ObjectClassAttribute + "=" +
-		config.GroupObjectClass + "))"
+	query := fmt.Sprintf("(&(%v=*%v*)(objectClass=%v))", config.GroupSearchAttribute, name, config.GroupObjectClass)
```

Comment:
we found contains queries to be too inefficient in large datasets. Can you change this to a startsWith query

Reply:
sure